Dz1

NikitaBarshchevskii opened the pull request:
# Домашнее задание №1

Прошу @martinkomitsky, @haseprogram, @kgrechin или @ivan-bir проверить его.

Что было сделано:
* Реализация функций в файлах `convertBytesToHuman.js`, `correctSentence.js`, `nonUniqueElements.js`;
* Написание тестов на функцию `convertBytesToHuman`;
* Написание дополнительных тестов для `correctSentence`, `nonUniqueElements`;

NikitaBarshchevskii committed:
Init commit

NikitaBarshchevskii committed:
Create pull_request_template.md

NikitaBarshchevskii committed:
Minor tweaks

NikitaBarshchevskii committed:
Minor tweaks

Ivan-Bir commented on convertBytesToHuman.js:
```diff
@@ -0,0 +1,37 @@
+/*
```
Привет! Добро пожаловать на курс
Я буду твоим ментором и буду рад ответить на любые вопросы по курсу

Ivan-Bir commented on convertBytesToHuman.js:
```diff
@@ -0,0 +1,37 @@
+  let counter = 0;
+  const units = ['B', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB'];
+
+  //Converting bytes with each iteration
```
Слишком очевидно, чтобы пояснять

Ivan-Bir commented on convertBytesToHuman.js:
```diff
@@ -0,0 +1,37 @@
+  }
+  
+  bytes = parseFloat(bytes.toFixed(2));
+  const str = bytes + " " + units[counter];
```
Лучшие использовать шаблонные строки `${var1} ${var2}`, это более гибко и безопасно

Ivan-Bir commented on convertBytesToHuman.test.js:
```diff
@@ -0,0 +1,38 @@
+  expect(convertBytesToHuman(1610612736)).toBe("1.5 GB")
+});
+
+// другая группа проверок
```
Лишний комент

Ivan-Bir commented on correctSentence.js:
```diff
@@ -0,0 +1,35 @@
+ */
+
+export default function correctSentence(text) {
+  //Getting rid of possible whitespaces
```
Не надо пояснять каждую строчку коментом. Коменты используем только чтобы объяснить что-то неоднозначное, сомнительное или сложное. Здесь и далее

Ivan-Bir commented on correctSentence.js:
```diff
@@ -0,0 +1,35 @@
+
+  //Capitalizing the first letter
+  if (text.length > 0) {
+    text = text[0].toUpperCase() + text.slice(1);
```
Шаблонные строки, помним. Здесь и далее

Ivan-Bir commented on correctSentence.test.js:
```diff
@@ -0,0 +1,10 @@
+  expect(correctSentence("Greetings, friends.")).toBe("Greetings, friends.")
+  expect(correctSentence("   greetings, friends   ")).toBe("Greetings, friends.")
+  expect(correctSentence("   hello.   ")).toBe("Hello.")
+  expect(correctSentence("")).toBe(".")
```
Это не то поведение, которое ожидается. Более ожидаемым будет пустая строка

Ivan-Bir commented on nonUniqueElements.js:
```diff
@@ -0,0 +1,47 @@
+
+  //filling in result with all non-unique elements
+  data.forEach(element => {
+    if(encounteredElements.get(element) > 1){
```
Нет пробела перед фигурной скобкой. Чтобы не отвлекаться на ошибки форматирования, можно для vscode использовать Prettier

Ivan-Bir commented on nonUniqueElements.js:
```diff
@@ -0,0 +1,47 @@
+    }
+  });
+
+  return result
```
Проверь здесь и везде ;

Ivan-Bir commented on nonUniqueElements.js:
```diff
@@ -0,0 +1,47 @@
+  const result = [];
+
+  //filling in a Map of all encountered elements and their number
+  data.forEach(element => {
```
Даже если это единственный параметр стрелочной функции, то его принято оборачивать в скобки

Ivan-Bir commented on convertBytesToHuman.js:
```diff
@@ -0,0 +1,37 @@
+    return false;
+  }
+
+  let counter = 0;
```
Переменную объявляем как можно ближе к месту её использования

NikitaBarshchevskii committed:
Refactor and Fixes

- Перемещено объявление переменной `counter` ближе к её использованию;
- Улучшена читаемость кода;
- Добавлено использование шаблонных строк;
- Скорректировано поведение `correctSentence` и `correctSentence.test` для пустой строки;

Ivan-Bir commented:
Не придраться. Апрув

martinkomitsky commented:
12!
